crates/nu-command/src/filters/where_.rs

```diff
@@ -34,9 +42,9 @@ not supported."#
                 (Type::Range, Type::Any),
             ])
             .required(
-                "row_condition",
-                SyntaxShape::RowCondition,
-                "Filter condition.",
+                "condition",
+                SyntaxShape::OneOf(vec![SyntaxShape::RowCondition, SyntaxShape::Closure(None)]),
```

Comment:
```suggestion
                SyntaxShape::OneOf(vec![
                    SyntaxShape::RowCondition,
                    SyntaxShape::Closure(Some(vec![SyntaxShape::Any])),
                ]),
```
The closure's signature matches `filter` and `each` like this.